src/goose.rs

```diff
@@ -1222,6 +1204,29 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        // Extract the base_url of the request made.
+                        let request_base_url = match Url::parse(&raw_request.url) {
+                            Ok(base_url) => base_url.to_string(),
+                            Err(e) => {
+                                error!("goose_send failed to parse requested URL: {}", e);
+                                std::process::exit(1);
+                            }
+                        };
+                        // If a request for the base_url was redirected, change base_url.
+                        if request_base_url == self.base_url.read().await.to_string() {
```

Comment:
It might be needed to disable that. i.e. if a test accidentally would redirect to IDMS auth we definitely don't want to load test IDMS. Or let's assume a load test accidentally forwards to google.com:

You would not want to accidentally loadtest google.com and get blamed for DDOS ...

Maybe this should only be done for `a subdomain of the original host`?

Again it is really important to understand how locust does it.

Reply:
Any suggestions on what to call this option? I was going to use `--no-redirect` but it wouldn't actually disable redirects, it would just disable the client following the redirect on subsequent requests. Perhaps `--no-follow` is therefor a better option name?

I consider it the load test writer's responsibility to keep their load test from load testing the wrong domains/paths/whatever.

Reply:
I ended up with `--no-sticky-follow` as it seems to be properly descriptive -- feedback and alternate suggestions welcome